packages/@react-stately/color/src/Color.ts

```diff
@@ -34,6 +35,11 @@ export class Color {
     return x;
   }
 
+  static parse(value: ColorInput): Color {
```

Comment:
Ah, I think this might be the reason that the onChange is firing so much, it keeps getting a new Color object

Reply:
I think `onChange` is firing frequently because I call it in `commitInputValue` whenever the HTML input's onchange is called and the string is valid.

```
  const setFieldInputValue = (value: string) => {
    setInputValue(value);
    value = value.trim();
    if (!value.length) { return; }
    if (!value.startsWith('#')) {
      value = `#${value}`;
    }
    try {
      const color = clampColor(value);
      if (color) {
        setColorValue(color);
      }
    } catch (err) {
      // ignore
    }
  };

  const commitInputValue = () => {
    setFieldInputValue(colorValue ? colorValue.toString('hex') : '');
  };
```

I previously checked how the input would behave when I had `setInputValue` in `commitInputValue` instead of `setFieldInputValue` and I wasn't getting desired behaviors in some scenarios. I can elaborate more once I start resolving this, if it still makes sense.

Reply:
sure, though just be aware that if you call setState from a useControlledState hook with the value that is already in there, it won't trigger the onChange argument, if you pass in a new object, it will, even if the new object represents the same color as the object before

Reply:
@snowystinger oh sorry I missed this question

> Is there a reason we need a clone of a given Color instance? I'd say lets just leave it to the constructor which takes a string and returns a new instance already, and if they already have a color, then they didn't need to parse it

I did this coz I keep getting the typescript error whenever I return just `value` if it's already a `Color` object. It's like it's not convinced the object I passed has a `value` attribute
<img width="972" alt="Screen Shot 2020-09-24 at 12 24 25 PM" src="https://user-images.githubusercontent.com/5743833/94190481-0527d200-fe61-11ea-8142-0fb7a4fcae13.png">

Reply:
I'll implement the `useColor` hook first and get back to this if this is still relevant

Reply:
Resolved by removing `Color.parse`